code/modules/organs/external/_external.dm

```diff
@@ -1305,33 +1332,42 @@ Note that amputating the affected organ does in fact remove the infection from t
 			for(var/atom/movable/implant in implants)
 				//large items and non-item objs fall to the floor, everything else stays
 				var/obj/item/I = implant
+				if(QDELETED(implant))
+					LAZYREMOVE(implants, implant)
+					continue
 				if(istype(I) && I.w_class < ITEM_SIZE_NORMAL)
+					if(istype(I, /obj/item/implant))
+						var/obj/item/implant/imp = I
+						imp.removed()
 					implant.forceMove(src)
 				else
+					//Dumpt the rest on the turf
 					LAZYREMOVE(implants, implant)
 					implant.forceMove(get_turf(src))
 
 			if(!ignore_children)
 				//Move our chilren limb into our contents
 				for(var/obj/item/organ/external/O in children)
-					victim.remove_organ(O, drop_organ = FALSE, ignore_children = TRUE)
+					victim.remove_organ(O, FALSE, FALSE, FALSE, in_place, update_icon)
 					if(QDELETED(O))
+						LAZYREMOVE(children, O)
 						continue
-					// if we didn't lose the organ we still want it as a child
-					O.forceMove(src)
-					LAZYDISTINCTADD(children, O)
-					O.parent = src
+					O.do_install(null, src, FALSE, update_icon, FALSE) //Forcemove the organ and properly set it up in our internal data
 
 			// Grab all the children internal organs
-			for(var/obj/item/organ/organ in internal_organs)
-				victim.remove_organ(organ, FALSE, FALSE, FALSE, in_place, update_icon) // Organ stays inside and connected
-				if(!QDELETED(organ))
-					organ.forceMove(src)
+			for(var/obj/item/organ/internal/organ in internal_organs)
+				victim.remove_organ(organ, FALSE, FALSE, FALSE, in_place, update_icon)
+				if(QDELETED(organ))
+					LAZYREMOVE(internal_organs, organ)
+					continue
+				organ.do_install(null, src, FALSE, update_icon, FALSE) //Forcemove the organ and properly set it up in our internal data
+
+	//Note that we don't need to change our own hierarchy when not removing from a mob
 
 	// Remove parent references
 	if(parent)
 		LAZYREMOVE(parent.children, src)
```

Comment:
That's something I considered. But, its not a very common case to have special behavior when a limb is outside a mob. I don't really have any applicable use cases for those in the codebase rn, so someone would probably just delete the proc since its unused. Limb behavior outside mobs currently is basically just ceasing any processing and just being in a proper hierarchy so icon updates and stuff works.

But, yeah, its a bit out of scope for this bugfix imho. But it would make sense to add that in the future.